### DISCUSSION TITLE
Week 7

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#50] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.
Week 7 has only just began with some new updates in the pipeline:
<br><br>

 ### UML class diagram
-----
✅ **[class diagram](https://github.com/SE-TINF22B6/CookHub/blob/main/Project_Doc/Diagrams/uml-class-diagram.drawio.svg) has been extended and now covers the whole application**
- implemented react components (mvc is now visible)
- adjusted relationships between classes ➜ dependency inversion is now possible
<br>

###  Design principles
------
✅ **2 design principles has been determined out of the project**
- [MVC](https://github.com/SE-TINF22B6/CookHub/wiki/Design-principles#mvc)
- [Dependency injection](https://github.com/SE-TINF22B6/CookHub/wiki/Design-principles#dependency-inversion)
➜ click to get to our wiki articles, which explain the principles and our usage of them in more detail

✅  **Team convention** 
- we decided to try sticking to both design principles
- added convention to our [team agreement](https://github.com/SE-TINF22B6/CookHub/wiki/Team-Agreement)
<br>

###  Definition of classes
------
✅ **The [definition of classes](https://github.com/SE-TINF22B6/CookHub/issues/63#issue-2004032035) in our application has been defined**
<br>

### Coding progression
-----
✅ **Added functionality to links on landing page** 
- Login, About us, FAQ's, Impressum ➜ redirects to their specific page <br>

✅ **Created impressum page**
- Impressum content  ➜ [impressum](https://github.com/SE-TINF22B6/CookHub/assets/113697513/2a781264-7ccd-4485-bbf5-d2e0978b9c56) is now available (snapshot)

✅ **Created FAQs page**
- FAQ's content ➜ [FAQ's](https://github.com/SE-TINF22B6/CookHub/assets/113697513/8bb5662c-306d-4da5-80df-bf998c34152c) is now available (snapshot)

✅ **Created About us page**
- About us content  ➜ [Team](https://github.com/SE-TINF22B6/CookHub/wiki/Team) now available in app & wiki

✅ **Added data base connection**
- Backend connects to a PostgreSQL data base when started
- Tests use an SQLite in-memory database to test the code of our repository patterns, to make sure that production data will not be influenced by tests

### OpenAI ChatGPT 
-----
✅ **Research & tests of more specific prompts to get better resolutions with [adventurize it Ⓒ](https://github.com/SE-TINF22B6/CookHub/wiki/adventurize-it-%E2%92%B8)** 

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by dolerich-hirnfiedler** [comment]
Dear CookHub Team, after reading your wiki pages for the mvc-pattern as well as for dependency inversion, I am sure you are on a good way with your project. It seems like you make good decisions regarding architecture and code style. I am thrilled to see how you will progress in the future.

**Comment by JonasEbbinghaus25** [comment]
A very detailed blog for this week. The UML diagram is very neat and detailed. I'm glad to hear that you were able to decide on two design principles and even documented them. The implementation of additional features such as linking Login, About Us, FAQs and Imprint, which now redirect to their respective pages, is a significant step forward. The application is taking shape. I look forward to seeing your team's progress in the coming weeks!

**Comment by AndresMasis** [comment]
HI

I think that implementing dependency inversion will really benefit your project. The facto of having the higher-level modules to rely on abstractions rather than specific implementations, allowing for flexibility, extensibility, and easier maintenance within a system. This is a great improvement.

In addition, I appreaciate the high level of detail in your blog. It really allows the reader to keep track on what you are doing.

Cheers from CookPit
